From 6be17375f9142017d34bd0b577cdf4e91225c135 Mon Sep 17 00:00:00 2001
From: Paul Zander <negril.nx+gentoo@gmail.com>
Date: Fri, 22 Nov 2024 19:46:05 +0100
Subject: [PATCH] don't include <ciso646> with c++17

[179/825] Building CXX object absl/log/CMakeFiles/log_globals.dir/globals.cc.o
In file included from
/var/tmp/paludis/dev-cpp-abseil-cpp-20240722.0/work/abseil-cpp-20240722.0/absl/hash/internal/hash.h:33,
from
/var/tmp/paludis/dev-cpp-abseil-cpp-20240722.0/work/abseil-cpp-20240722.0/absl/hash/hash.h:85,
from
/var/tmp/paludis/dev-cpp-abseil-cpp-20240722.0/work/abseil-cpp-20240722.0/absl/log/globals.cc:29:
/usr/lib/gcc/x86_64-pc-linux-gnu/15/include/g++-v15/ciso646:46:4: warning:
#warning "<ciso646> is deprecated in C++17, use <version> to detect
implementation-specific macros" [-Wcpp]
46 | #  warning "<ciso646> is deprecated in C++17, use <version> to detect
implementation-specific macros"
      |    ^~~~~~~

Signed-off-by: Paul Zander <negril.nx+gentoo@gmail.com>

diff --git a/absl/hash/internal/hash.h b/absl/hash/internal/hash.h
index 03bf183..3222be3 100644
--- a/absl/hash/internal/hash.h
+++ b/absl/hash/internal/hash.h
@@ -27,7 +27,7 @@
 #include "absl/base/config.h"
 
 // For feature testing and determining which headers can be included.
-#if ABSL_INTERNAL_CPLUSPLUS_LANG >= 202002L
+#if ABSL_INTERNAL_CPLUSPLUS_LANG >= 201703L
 #include <version>
 #else
 #include <ciso646>
-- 
2.47.0

